dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.RegExMatch;
+import com.dotmarketing.util.UtilMethods;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ *
+ * @author nollymar
+ */
+public class LuceneQueryDateTimeFormatter {
+
+
+    //Date formats supported by dotCMS in lucene queries
+    private static final Map<String, String> LUCENE_DATE_TIME_FORMAT_PATTERN = new LinkedHashMap<String, String>(){
+        {
+            put("MM/dd/yyyy hh:mm:ssa",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2}:\\d{1,2}(?:AM|PM|am|pm))\\\"?");
+
+            put("MM/dd/yyyy hh:mm:ss a",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2}:\\d{1,2}\\s+(?:AM|PM|am|pm))\\\"?");
+
+            put("MM/dd/yyyy hh:mm a",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2}\\s+(?:AM|PM|am|pm))\\\"?");
+
+            put("MM/dd/yyyy hh:mma",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2}(?:AM|PM|am|pm))\\\"?");
+
+            put("MM/dd/yyyy HH:mm:ss",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2}:\\d{1,2})\\\"?");
+
+            put("MM/dd/yyyy HH:mm",
+                    "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4}\\s+\\d{1,2}:\\d{1,2})\\\"?");
+
+            put(datetimeFormat.getPattern(),
+                    "\\\"?(\\d{4}\\d{2}\\d{2}\\d{2}\\d{2}\\d{2})\\\"?");
+
+            put(dateFormat.getPattern(), "\\\"?(\\d{4}\\d{2}\\d{2})\\\"?");
+
+            put("MM/dd/yyyy", "\\\"?(\\d{1,2}/\\d{1,2}/\\d{4})\\\"?");
+
+            put("hh:mm:ssa", "\\\"?(\\d{1,2}:\\d{1,2}:\\d{1,2}(?:AM|PM|am|pm))\\\"?");
+
+            put("hh:mm:ss a", "\\\"?(\\d{1,2}:\\d{1,2}:\\d{1,2}\\s+(?:AM|PM|am|pm))\\\"?");
+
+            put("HH:mm:ss", "\\\"?(\\d{1,2}:\\d{1,2}:\\d{1,2})\\\"?");
+
+            put("hh:mma", "\\\"?(\\d{1,2}:\\d{1,2}(?:AM|PM|am|pm))\\\"?");
+
+            put("hh:mm a", "\\\"?(\\d{1,2}:\\d{1,2}\\s+(?:AM|PM|am|pm))\\\"?");
+
+            put("HH:mm", "\\\"?(\\d{1,2}:\\d{1,2})\\\"?");
+        }
+    };
+
+    private static final List<String> LUCENE_TIME_FORMAT_PATTERNS = new ArrayList<>(
+            Arrays.asList("hh:mm:ssa", "hh:mm:ss a", "HH:mm:ss", "hh:mma", "hh:mm a", "HH:mm"));
+    /**
+     *
+     * @param query
+     * @return
+     */
+    static String findAndReplaceQueryDates(String query) {
+        query = RegEX.replaceAll(query, " ", "\\s{2,}");
+
+        //delete additional blank spaces on date range
+        if(UtilMethods.contains(query, "[ ")) {
+            query = query.replace("[ ", "[");
+        }
+
+        if(UtilMethods.contains(query, " ]")) {
+            query = query.replace(" ]", "]");
+        }
+
+        String clausesStr = RegEX.replaceAll(query, "", "[\\+\\-\\(\\)]*");
+        String[] tokens = clausesStr.split(" ");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'tokens' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)